### DISCUSSION TITLE
Server error 111 after upgrade to v3

### DISCUSSION THREAD

**Discussion by jburgon** [netbox-community/netbox#7222] [Q&A]
I am running netbox in a docker container in unraid. It updated to version 3.0.3-dev from version 2 and now it shows a server error. Any ideas? Thanks

<class 'redis.exceptions.ConnectionError'>

Error 111 connecting to localhost:6379. Connection refused.

Python version: 3.9.7
NetBox version: 3.0.3-dev

**Comment by candlerb** [comment]
1. Version 2 what?  Upgrades to 3.x are only supported from 2.11.x
2. Don't ever install a dev version of Netbox unless you're prepared to throw away your data and start again.  Upgrades from dev versions to any other version (dev or production) are not supported.  If you want to try out a dev version, then you should replicate your database from a production version, but you should keep your production version running and make any important data changes there.
3. Clearly the redis server isn't starting.  That means it must be a problem with your docker container or docker-compose config

What container image are you running, and where did you get it from?  Are you using docker-compose?

If you're using assets published by the [netbox-docker](https://github.com/netbox-community/netbox-docker) project, then you probably want to raise this separately as an issue there.  But first I suggest that you try to reproduce this problem with a fresh docker container, before you attempt upgrading a database from a previous version.

**Comment by w4rh0und** [comment]
Hi jburgon

In my case it was an internal communication with redis. 

I've ran a simple telnet localhost 6379 
Trying 127.0.1.1...
telnet: Unable to connect to remote host: Connection refused

Ran telnet 127.0.0.1 6379 .... worked 

I've changed the config for redis 
under /etc/redis/redis.conf 
from: bind 127.0.0.1 
to: bind 127.0.0.1 localhost

Worked like a charm after 

Hope it helps in your case as well

**Reply by candlerb** [reply]
Thank you, that makes a lot of sense.  It means localhost is wrongly resolving to 127.0.1.1 instead of 127.0.0.1.  Correcting the localhost entry in `/etc/hosts` would have been another solution.

**Comment by jburgon** [comment]
Thanks for the input. My previous version was 2.11 (not sure exactly which one but it was the latest). I'm running this in docker in unraid using this image: https://hub.docker.com/r/pitkley/netbox. I simply updated the container because an update was available and it updated to a dev version. I didn't ask for a dev version nor would I have chosen it if presented with that option.

The file /etc/redis/redis.conf doesn't exist in my container. Should I be looking somewhere else? I might switch to the linuxserver container so if I have to rebuild everything it's fine.

**Reply by candlerb** [reply]
> The file /etc/redis/redis.conf doesn't exist in my container. Should I be looking somewhere else? I might switch to the linuxserver container so if I have to rebuild everything it's fine.

I don't run netbox in a container so I can't speak authoritatively.

However, if you're using the image built from the [netbox-docker repo](https://github.com/netbox-community/netbox-docker) then I believe you're supposed to run it with docker-compose; that will create separate containers for netbox, redis, and postgres, and configure them to talk to each other. See the [docker-compose.yml](https://github.com/netbox-community/netbox-docker/blob/release/docker-compose.yml) file.  That would also explain why there's no redis.conf in the netbox container; it belongs in the redis container.

I'm afraid I don't know what you mean by "the linuxserver container".

**Comment by ryanmerolle** [comment]
I avoid linuxserver containers unless I need their docker volume file permission tweaks or a software is not containerized by the main project.  They generally do not support their projects to the degree the actual maintainers of a project would.

The update of netbox-docker is pretty straight forward if you follow the release notes.  First of all, if you originally cloned the netbox-repo, please be aware that the previous netbox-docker 1.2.0 only supports netbox v2.11.  I would pull down tag/release 1.3.1 and be sure to read the release notes for 1.3.0 and 1.3.1 since there is some extra steps you need to take.  https://github.com/netbox-community/netbox-docker/releases
